saphir/src/file/cache.rs

```diff
@@ -0,0 +1,301 @@
+use crate::{
+    error::SaphirError,
+    file::{compress_file, middleware::PathExt, Compression, Encoder, File, FileCursor, FileInfo, FileStream, SaphirFile, MAX_BUFFER},
+};
+use futures::{
+    io::{AsyncRead, AsyncSeek, Cursor},
+    AsyncReadExt, AsyncSeekExt, Future,
+};
+use mime::Mime;
+use std::{
+    collections::HashMap,
+    io,
+    io::SeekFrom,
+    path::PathBuf,
+    pin::Pin,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    task::{Context, Poll},
+};
+use tokio::sync::RwLock;
+
+type InnerCache = Arc<RwLock<HashMap<(String, Compression), Vec<u8>>>>;
+#[derive(Clone)]
+pub struct FileCache {
+    inner: InnerCache,
+    max_file_size: u64,
+    max_capacity: u64,
+    current_size: Arc<AtomicU64>,
```

Comment:
**Atomic**U64 does not need to be wrapped in an Arc (**Atomic**RefCount) since it is already **Atomic**